trace-sdk/src/test/java/io/bitrise/trace/data/collector/cpu/CpuStatTest.java

```diff
@@ -0,0 +1,43 @@
+package io.bitrise.trace.data.collector.cpu;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Unit tests for {@link CpuUsageData.CpuStat}
+ */
+public class CpuStatTest {
+    final CpuUsageData.CpuStat cpuStat = new CpuUsageData.CpuStat(
+            1, 2, 3, 4, 5, 6, 7, 8 );
+
+    @Test
+    public void toString_shouldMatch() {
+        final String expected = "CpuStat{user=1.0, system=3.0, nice=2.0, idle=4.0, ioWait=5.0, irq=6.0, softIrq=7.0, steal=8.0}";
+        assertEquals(expected, cpuStat.toString());
+    }
+
+    @Test
+    public void equals_sameObject() {
+        assertTrue(cpuStat.equals(cpuStat));
+    }
+
+    @Test
+    public void equals_notCpuStat() {
+        assertFalse(cpuStat.equals("cpu_stat"));
+    }
+
+    @Test
+    public void equals_differentCpuStat() {
+        final CpuUsageData.CpuStat differentCpuStat = new CpuUsageData.CpuStat(
+                1, 2, 3, 4, 5, 6, 7, 80 );
+        assertFalse(cpuStat.equals(differentCpuStat));
+    }
+
+    @Test
+    public void hashCode_shouldMatch() {
```

Comment:
Following the logic above, shouldn't we test instead, that hash code of the same object is equal? (instead of a magic number)

Reply:
Sure - I can update my code to test the same object provides the same hash code :)